x/axelarnet/message_handler.go

```diff
@@ -21,17 +22,38 @@ import (
 
 // Fee is used to pay relayer for executing cross chain message
 type Fee struct {
-	Amount    string `json:"amount"`
-	Recipient string `json:"recipient"`
+	Amount          string  `json:"amount"`
+	Recipient       string  `json:"recipient"`
+	RefundRecipient *string `json:"refund_recipient"`
 }
 
-func (f Fee) validate(ctx sdk.Context, n types.Nexus, b types.BankKeeper, token sdk.Coin, msgType nexus.MessageType) error {
-	amt, ok := sdk.NewIntFromString(f.Amount)
-	if !ok || amt.LTE(sdk.ZeroInt()) {
+func (f Fee) validateBasic() error {
+	amount, ok := sdk.NewIntFromString(f.Amount)
+	if !ok || !amount.IsPositive() {
 		return fmt.Errorf("invalid fee amount")
 	}
 
-	afterFee := token.Amount.Sub(amt)
+	if _, err := sdk.AccAddressFromBech32(f.Recipient); err != nil {
+		return err
+	}
+
+	if f.RefundRecipient != nil {
+		// The refund recipient is an address of the chain where the request is originally initiated from,
+		// and therefore we cannot validate much of it.
+		if err := utils.ValidateString(*f.RefundRecipient); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (f Fee) validate(ctx sdk.Context, n types.Nexus, b types.BankKeeper, token sdk.Coin, msgType nexus.MessageType) error {
+	if err := f.validateBasic(); err != nil {
+		return err
+	}
+
+	afterFee := token.Amount.Sub(funcs.MustOk(sdk.NewIntFromString(f.Amount)))
 	switch msgType {
```

Comment:
I couldn't select the `case nexus.TypeGeneralMessageWithToken:` directly, so this comment is slightly off the correct line

```suggestion
	case nexus.TypeGeneralMessageWithToken:
		if !afterFee.IsPositive() {
			return fmt.Errorf("transfer amount must be non-zero after fees are deducted")
}
```

Reply:
lol didn't even read this part of the code and didn't realize that there are more places where `IsPositive` and `IsNegative` should be used. Will do.